source/patterns/@aws-solutions-constructs/aws-route53-apigateway/README.md

```diff
@@ -0,0 +1,122 @@
+# aws-route53-apigateway module
+<!--BEGIN STABILITY BANNER-->
+
+---
+
+![Stability: Experimental](https://img.shields.io/badge/stability-Experimental-important.svg?style=for-the-badge)
+
+> All classes are under active development and subject to non-backward compatible changes or removal in any
+> future version. These are not subject to the [Semantic Versioning](https://semver.org/) model.
+> This means that while you may use them, you may need to update your source code when upgrading to a newer version of this package.
+
+---
+<!--END STABILITY BANNER-->
+
+| **Reference Documentation**:| <span style="font-weight: normal">https://docs.aws.amazon.com/solutions/latest/constructs/</span>|
+|:-------------|:-------------|
+<div style="height:8px"></div>
+
+| **Language**     | **Package**        |
+|:-------------|-----------------|
+|![Python Logo](https://docs.aws.amazon.com/cdk/api/latest/img/python32.png) Python|`aws_solutions_constructs.aws_route53_apigateway`|
+|![Typescript Logo](https://docs.aws.amazon.com/cdk/api/latest/img/typescript32.png) Typescript|`@aws-solutions-constructs/aws-route53-apigateway`|
+|![Java Logo](https://docs.aws.amazon.com/cdk/api/latest/img/java32.png) Java|`software.amazon.awsconstructs.services.route53apigateway`|
+
+## Overview
+
+This AWS Solutions Construct implements an Amazon Route 53 connected to a configured Amazon API Gateway REST API.
+
+Here is a minimal deployable pattern definition in Typescript:
+
+``` typescript
+import * as api from '@aws-cdk/aws-apigateway';
+import * as lambda from "@aws-cdk/aws-lambda";
+import * as route53 from "@aws-cdk/aws-route53";
+import { Route53ToApigateway } from '@aws-solutions-constructs/aws-route53-apigateway';
+
+const lambdaProps: lambda.FunctionProps = {
+    code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+    runtime: lambda.Runtime.NODEJS_12_X,
+    handler: 'index.handler'
+};
+
+const lambdaFunction = new lambda.Function(this, 'LambdaFunction', lambdaProps);
+
+const apiGatewayProps: api.LambdaRestApiProps = {
+    handler: lambdaFunction,
+    endpointConfiguration: {
+        types: [api.EndpointType.REGIONAL]
+    },
+    defaultMethodOptions: {
+        authorizationType: api.AuthorizationType.NONE
+    }
+};
```

Comment:
Couple issues here:

* This really complicates the issue - we're 20 lines into the code and there's nothing specific to the construct. 
* We only accept an existing Hosted Zone because we believe that devs won't have the rights to create and destroy public hosted zones dynamically. So let's wrap an HZ already in the account.

Let's skip over all this and just say:

```
// The construct requires an existing REST API, this can be created in raw CDK or extracted
// from a previously instantiated construct that created an API Gateway REST API
const existingRestApi = previouslyCreatedApigatewayToLambdaConstruct.apiGateway;

const ourHostedZone = route53.HostedZone.fromLookup(this, 'HostedZone', {
    domainName: "example.com",
  });

// then instantiate the construct
```

And, yes, it's domainName in fromLookup and zoneName everywhere else. I bet the CDK folks wish they could get a do over on that. :-)